File: plugins/org.eclipse.osee.web/src/app/shared-components/services/branch.service.ts
/*********************************************************************
 * Copyright (c) 2022 Boeing
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Boeing - initial API and implementation
 **********************************************************************/
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { apiURL } from 'src/environments/environment';
import { branch } from '../../types/branches/branch';

@Injectable({
  providedIn: 'root'
})
export class BranchService {

  constructor (private http: HttpClient) { }
  
  public getBranches(type: string,category:string) {
    return this.http.get<branch[]>(apiURL+`/orcs/branches/${type}/category/${category}`);
  }
}

File: plugins/org.eclipse.osee.framework.core/src/org/eclipse/osee/framework/core/data/BranchService.java
/*********************************************************************
 * Copyright (c) 2025 Boeing
 *
 * This program and the accompanying materials are made
 * available under the terms of the Eclipse Public License 2.0
 * which is available at https://www.eclipse.org/legal/epl-2.0/
 *
 * SPDX-License-Identifier: EPL-2.0
 *
 * Contributors:
 *     Boeing - initial API and implementation
 **********************************************************************/
package org.eclipse.osee.framework.core.data;

/**
 * @author Donald G. Dunne
 */
public interface BranchService {

   /**
    * Toggle branch as favorite for current user
    */
   void toggleFavoriteBranch(BranchId branch);

   /**
    * @return true if branch is marked as favorite for current users. <br/>
    * <br/>
    * Cached for efficiency as this is called thousands of times to sort favorites. Cache is updated when User artifact
    * is changed (eg: transaction in User art is different)
    */
   boolean isFavoriteBranch(BranchId branch);

}
